### PR TITLE
Three Ninjas Gamemode

### PR DESCRIPTION
**Pull request by BurgerLUA**
# Overview
Adds a new antag type, "Lesser Ninja." Lesser ninjas are like regular ninjas, except they don't have a teleportation module, but have upgraded rigsuit modules such as a better low-cooldown low-cost robust door hacking tool that can remotely hack multiple doors within 8 tiles. However, in order to modify a door's properties, you must be next to the door.

Adds the gamemodes "Triple Threat" and "Quadruple Threat".  Triple threat involves three lesser ninjas, while quadruple threat involves three lesser ninjas and one regular ninja. Currently they are not in secret rotation (as this is strictly a config option handled by skull and arrow), however both gamemodes can be voted for.

Adds several tracking beacons (these are basically teleporter destinations for ninjas, the hand teleporter, and the teleportation room) around the station in sane areas. Modified the code so that tracking beacons can be turned off by activating them in hand.

The existing door hacking tool's RNG was partially removed. Doors can be hacked in 30 to 35 seconds, and it stores 7 doors.

Ninja spawning outfits and equipment was changed to be better. Ninjas now get better belts and black webbing to hold items. They also get a better jumpsuit and have a ski mask just because.

Removed redundant ninja modules from the uplink. The modules removed exist for both normal ninjas and lesser ninjas.

### Ninja Spawn
Ninja

### Teleport Locations
Teleport Location

## Feedback Thread
https://forums.aurorastation.org/topic/11220-three-ninjas-gamemode/

## TODO
- [x] Initial Testing
- [x] Initial PR
- [x] Further Tweaking and Improvements
- [x] Feedback and Tweaking
- [ ] Final Testing

### CODE REVIEW

**Comment by BurgerLUA** [comment]
!review

File: code/game/antagonist/outsider/ninja_minor.dm [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	ninja_objective.owner = ninja
+	ninja.objectives += ninja_objective
+
+/datum/antagonist/ninja_minor/greet(var/datum/mind/player)
+
+	if(!..())
+		return 0
+	var/directive = generate_ninja_directive("heel")
+	player.store_memory("<B>Directive:</B> <span class='danger'>[directive]</span><br>")
+	player << "<b>Remember your directive:</b> [directive]."
```
Comment (skull132) [inline code/game/antagonist/outsider/ninja_minor.dm]:
```suggestion
	to_chat(player, "<b>Remember your directive:</b> [directive].")
```

File: code/game/antagonist/outsider/ninja_minor.dm [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	if(istype(player.back,/obj/item/weapon/rig))
+		var/obj/item/weapon/rig/rig = player.back
+		if(rig.air_supply)
+			player.internal = rig.air_supply
+
+	spawn(10)
+		if(player.internal)
+			player.internals.icon_state = "internal1"
+		else
+			player << "<span class='danger'>You forgot to turn on your internals! Quickly, toggle the valve!</span>"
```
Comment (skull132) [inline code/game/antagonist/outsider/ninja_minor.dm]:
```suggestion
			to_chat(player, "<span class='danger'>You forgot to turn on your internals! Quickly, toggle the valve!</span>")
```

File: code/game/antagonist/outsider/ninja_minor.dm [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+	if(ninjasuit)
+		ninjasuit.toggle_seals(src,1)
+		ninjasuit.seal_delay = initial(ninjasuit.seal_delay)
+
+	if(istype(player.back,/obj/item/weapon/rig))
+		var/obj/item/weapon/rig/rig = player.back
+		if(rig.air_supply)
+			player.internal = rig.air_supply
+
+	spawn(10)
```
Comment (skull132) [inline code/game/antagonist/outsider/ninja_minor.dm]: This should be a timer callback.

**Comment by skull132** [comment]
Unaddressed changes for close to/over 1 month. Closing.

**Pull request closed without merging**